File: re_leet_code/two_sum.py
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hash_map = {}
        for index, value in enumerate(nums):
            require_value = target - value
            if require_value in hash_map:
                return [index, hash_map[require_value]]
            else:
                hash_map[value] = index


nums = [2, 7, 11, 15]
target = 9
solution = Solution().twoSum(nums=nums, target=target)
print(solution)

File: Go/two_sum.go
package main

func TwoSum(nums []int, target int) []int {
	seen := make(map[int]int)
	for index, value := range nums {
		required := target - value
		if idx, ok := seen[required]; ok {
			return []int{index, idx}
		}
		seen[value] = index
	}
	return nil
}
